torchtext/experimental/models/roberta_model.py

```diff
@@ -0,0 +1,74 @@
+import os
+import torch
+import torch.nn as nn
+from torchtext.experimental.modules import TransformerEncoder
+from .roberta_transform import RobertaTransform
+from torchtext.utils import download_from_url, extract_archive, files_exist, load_args_from_json
+
+
+class RobertaModel(nn.Module):
+    """Contain a Roberta encoder model."""
+
+    def __init__(self, args, encoder):
+        super(RobertaModel, self).__init__()
+        self.args = args
+        self.encoder = encoder
+        self.transform = None
+
+    @classmethod
+    def build_model(self, args):
+        encoder = TransformerEncoder(args.ntoken, args.embed_dim, args.nhead,
+                                     args.feedforward_dim, args.nlayers, args.dropout)
+        return cls(args, encoder)
+
+    def load_transform(self, directory='./', tokenizer_file="sentencepiece.bpe.model", vocab_file='vocab.txt'):
+        self.transform = RobertaTransform.from_pretrained(directory=directory,
+                                                          tokenizer_file=tokenizer_file,
+                                                          vocab_file=vocab_file)
+
+    # [TODO] use torch.hub.load_state_dict_from_url
+    @classmethod
+    def from_pretrained(cls, args, directory='./', checkpoint_file="model.pt"):
+        encoder = TransformerEncoder(args.ntoken, args.embed_dim, args.nhead,
+                                     args.feedforward_dim, args.nlayers, args.dropout)
+        filepath = os.path.join(directory, checkpoint_file)
+        encoder.load_state_dict(torch.load(filepath))
+        return cls(args, encoder)
+
+    def forward(self, src):
+        src = self.bert_embed(src)
+        output = self.transformer_encoder(src)
+        return output
+
+    def encode(self, input_src: str) -> torch.Tensor:
+        return torch.tensor(self.transform(input_src), dtype=torch.long)
+
+
+# [TODO] Add torch.hub support
+# [TODO] Download file from manifold
+def xlmr_base_model(directory='./', checkpoint_file='model.pt', args_file='args.json',
```

Comment:
Why do we need a `directory` file, instead of just specifying full paths for the `checkpoint` / `args` / `tokenizer` / `vocab`? What if they are not all in the same directory?